cmd/oras/push.go

```diff
@@ -101,7 +104,13 @@ func runPush(opts pushOptions) error {
 	copyOptions.PreCopy = display.StatusPrinter("Uploading", opts.Verbose)
 	copyOptions.OnCopySkipped = display.StatusPrinter("Exists   ", opts.Verbose)
 	copyOptions.PostCopy = display.StatusPrinter("Uploaded ", opts.Verbose)
-	desc, err := packManifest(ctx, store, annotations, &opts)
+
+	var desc ocispec.Descriptor
+	if opts.artifactType == "" {
```

Comment:
Might be out of scope, looks like `packOCIManifest` and `packORASArtifactManifest` share the same input and output, and their logics are very similar. Maybe we can apply strategy pattern to it.

Reply:
And also the code used to pack an oras artifact is `attach.go` is duplicated with`packORASArtifactManifest` here.